woocommerce/woocommerce-ios
# Shipping Labels: Create/activate new package on Add New Package screen

Part of: #4745

## Description

This makes the Done button functional on the Add New Package screen in the shipping labels flow.

When the Done button is tapped, the new custom or service package is created/activated remotely and the screen is dismissed. The new package is selected for the shipping label and is listed in the Package Selected screen (the screen before the Add New Package screen if the store has existing packages).

Note: If the API calls fail, the Add New Package screen is not dismissed. However, this PR does not include any user-facing error messaging. We should show a notice if there is an error, but it may take some work to get the notice presenter working with SwiftUI so I haven't included that here.

## Changes

Networking:

* Updates the `createPackage` endpoint in `ShippingLabelRemote` to support activating a service package.
* Updates the models for `ShippingLabelPackagesResponse` and `ShippingLabelPredefinedOption` to save the provider ID (e.g. "usps" or "dhlexpress") for predefined options. This is used by the `createPackage` endpoint when activating a service package.
* Updates the Fakes and unit tests with those changes.

Yosemite:

* Updates the Shipping Label Store and Action to support the changes to the `createPackage` endpoint.

UI:

* Makes the Done button functional in `ShippingLabelAddNewPackage`:
   * Uses the state property `isSyncing` to display an activity indicator and disable the done button while the API calls are in progress.
   * If the API calls are successful, it dismisses the view.
* Adds helpers to `ShippingLabelAddNewPackageViewModel` for the Done button actions (`createCustomPackage` and `activateServicePackage`). It gets the new custom/service package from the child view, creates/activates the package remotely, syncs the package list from remote, and passes the new package and updated package list back to the Package List / Package Details.

<img src="https://user-images.githubusercontent.com/8658164/132456658-1f7f9aeb-1383-4716-afeb-5850b522a18e.gif" width="300px">


## Testing

1. Make sure you have configured the WooCommerce WCShip plugin.
2. Launch the app in debug mode (so the feature flag is enabled).
3. Open an order detail that is eligible for the shipping label.
4. Tap the button "Create Shipping Label.”
5. Confirm "Ship From" and "Ship To.”
6. In the package details section, tap on “Package Selected.”
7. If you already have existing packages on your store, you will see the Package Selected screen. Tap on "Create new package" at the bottom of the screen.
8. On the Add New Package screen, fill in the Custom Package form or select a Service Package.
9. Tap the Done button and notice there is an activity indicator and the button is disabled, and when it finishes you are taken back to the previous screen where the new package is selected.

## Submitter Checklist

Update release notes:
